views/js/pciCreator/ims/likertScoreInteraction/imsPciCreator.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016-2024 (original work) Open Assessment Technologies SA;
+ *
+ */
+define([
+    'lodash',
+    'likertScoreInteraction/creator/widget/Widget',
+    'tpl!likertScoreInteraction/creator/tpl/markup'
+], function(_, Widget, markupTpl){
+    'use strict';
+
+    const _typeIdentifier = 'likertScoreInteraction';
+
+    return {
+        /**
+         * (required) Get the typeIdentifier of the custom interaction
+         *
+         * @returns {String}
+         */
+        getTypeIdentifier: function() {
+            return _typeIdentifier;
+        },
+
+        /**
+         * (required) Get the widget prototype
+         * Used in the renderer
+         *
+         * @returns {Object} Widget
+         */
+        getWidget: function() {
+            return Widget;
+        },
+
+        /**
+         * (optional) Get the default properties values of the pci.
+         * Used on new pci instance creation
+         *
+         * @returns {Object}
+         */
+        getDefaultProperties: function() {
+            return {
+                level: 5,
+                'label-min': 'min',
+                'label-max': 'max',
+                icons: false,
+                numbers: false,
+            };
+        },
+
+        /**
+         * (optional) Callback to execute on the
+         * Used on new pci instance creation
+         *
+         * @returns {Object}
+         */
+        afterCreate: function(pci) {
+            //always set the NONE response processing mode to likert score
+            pci.getResponseDeclaration().setTemplate('NONE');
```

Comment:
**question:** Is this really needed? The response declaration does not seem to prevent from using the regular response processing.